### ISSUE TITLE
build: Validate requirements and hints are handlled appropriately.

### ISSUE THREAD

**Issue by qwofford** [lanl/BEE#365]
Fix if they aren't. Looks like scenario with both hints and requirements might prefer hints over requirements, which is wrong.

https://github.com/lanl/BEE_Private/blob/761dbc6aad9fd1d96e3ecb84e3fe1f636bb8aa76/src/beeflow/common/build/container_drivers.py#L183-L187

This will need to be corrected throughout the file. Potential code movement opportunity.

**Comment by pagrubel** [comment]
If you want requirements to supercede hints just ignore hints if they are both set, I think this does that:
```
if req_addr:
    task_addr = req_addr
elif hint_addr:
    task_addr = hint_addr
```

**Comment by qwofford** [comment]
haha yes, this is right. Must have been a long day when I made this mistake...

**Comment by qwofford** [comment]
btw, I re-made the truth table that I was talking about on the call. It was just bad logic.

**Comment by qwofford** [comment]
Resolved with #370
